spec/mem/pointer.md

```diff
@@ -129,35 +129,52 @@ pub enum PtrType {
 
 
 ```rust
-impl SizeStrategy {
+impl LayoutStrategy {
     pub fn is_sized(self) -> bool {
-        matches!(self, SizeStrategy::Sized(_))
+        matches!(self, LayoutStrategy::Sized(_, _))
     }
 
     /// Returns the size when the type must be statically sized.
-    pub fn expect_sized(self, msg: &str) -> Size {
+    pub fn expect_size(self, msg: &str) -> Size {
         match self {
-            SizeStrategy::Sized(size) => size,
-            _ => panic!("expect_sized called on unsized type: {msg}"),
+            LayoutStrategy::Sized(size, _) => size,
+            _ => panic!("expect_size called on unsized type: {msg}"),
+        }
+    }
+
+    /// Returns the alignment when the type must be statically sized.
+    pub fn expect_align(self, msg: &str) -> Align {
+        match self {
+            LayoutStrategy::Sized(_, align) => align,
+            _ => panic!("expect_align called on unsized type: {msg}"),
         }
     }
 
     /// Computes the dynamic size, but the caller must provide compatible metadata.
-    pub fn compute(self, meta: Option<PointerMeta>) -> Size {
+    pub fn compute_size(self, meta: Option<PointerMeta>) -> Size {
         match (self, meta) {
-            (SizeStrategy::Sized(size), None) => size,
+            (LayoutStrategy::Sized(size, _), None) => size,
             // FIXME(UnsizedTypes): We need to assert that the resulting size isn't too big.
```

Comment:
Not "assert", we need to raise UB if it is too big. This will require access to a target machine instance.